src/client/auth/abstractAuthentication.ts

```diff
@@ -19,6 +19,20 @@ export interface AuthenticationMethod {
 	): Promise<AuthenticationResult>
 }
 
+export interface AuthenticationMethodUN {
+	TYPE: string
+	getTokenProof(request: AuthenticateInterface): Promise<AuthenticationResult>
```

Comment:
@nicktaras interface AuthenticationMethod use same method name, but different input params. better change method name or make same params. This difference can lead to errors, better avoid that.

Reply:
Can we confirm the changes needed. 

I'd added the Un interface to simplify the UN export class SignedUNChallenge extends AbstractAuthentication implements AuthenticationMethodUN

Which had a different interface to AuthenticationMethod whilst developing the changes for multi-auth. 

I'll take a further look at this today, in case I've not understood the first time reading this correctly.

Reply:
I mean AuthenticationMethod and AuthenticationMethodUN have mathod **getTokenProof**, but those methods have different input params, it can lead to errors